spark/src/test/scala/org/apache/spark/sql/delta/DeltaTableFeatureSuite.scala

```diff
@@ -490,6 +491,52 @@ class DeltaTableFeatureSuite
     }
   }
 
+  test("drop table feature works with managed commits") {
+    val table = "tbl"
+    withTable(table) {
+      spark.range(0).write.format("delta").saveAsTable(table)
+      val log = DeltaLog.forTable(spark, TableIdentifier(table))
+      val featureName = TestRemovableReaderWriterFeature.name
+      assert(!log.update().protocol.readerAndWriterFeatureNames.contains(featureName))
+
+      // Add managed commit table feature to the table
+      CommitOwnerProvider.registerBuilder(InMemoryCommitOwnerBuilder(batchSize = 100))
+      val tblProperties1 = Seq(s"'${DeltaConfigs.MANAGED_COMMIT_OWNER_NAME.key}' = 'in-memory'")
+      sql(buildTablePropertyModifyingCommand(
+        "ALTER", targetTableName = table, sourceTableName = table, tblProperties1))
+
+      // Add TestRemovableReaderWriterFeature to the table in unbackfilled delta files
+      val tblProperties2 = Seq(s"'$FEATURE_PROP_PREFIX$featureName' = 'supported', " +
+        s"'delta.minWriterVersion' = $TABLE_FEATURES_MIN_WRITER_VERSION, " +
+        s"'${TestRemovableReaderWriterFeature.TABLE_PROP_KEY}' = 'true'")
+      sql(buildTablePropertyModifyingCommand(
+        "ALTER", targetTableName = table, sourceTableName = table, tblProperties2))
+      assert(log.update().protocol.readerAndWriterFeatureNames.contains(featureName))
+
+      // Disable feature on the latest snapshot
+      val tblProperties3 = Seq(s"'${TestRemovableReaderWriterFeature.TABLE_PROP_KEY}' = 'false'")
+      sql(buildTablePropertyModifyingCommand(
+        "ALTER", targetTableName = table, sourceTableName = table, tblProperties3))
+
+      val tableFeature =
+        TableFeature.featureNameToFeature(featureName).get.asInstanceOf[RemovableFeature]
+      assert(tableFeature.historyContainsFeature(spark, log.update()))
+
+      // Dropping feature should fail because the feature still has traces in deltas.
```

Comment:
Even without the fix in this PR, the DROP FEATURE will fail as expected as we trigger a checkpoint for feature drop and that backfills the commit?

Reply:
That's right. DROP FEATURE will still fail because checkpointing ensures there are no unbackfilled files.